qlty-check/src/planner/config_files.rs

```diff
@@ -148,3 +175,614 @@ pub fn plugin_configs(planner: &Planner) -> Result<HashMap<String, Vec<PluginCon
 
     Ok(configs)
 }
+
+pub fn compute_config_staging_operations(planner: &Planner) -> Result<Vec<ConfigOperation>> {
+    let plugins = enabled_plugins(planner)?;
+    let all_config_paths = collect_all_config_paths(&plugins);
+
+    let mut operations = Vec::new();
+    operations.extend(repository_config_operations(planner, &all_config_paths)?);
+    operations.extend(exported_config_operations(planner, &plugins)?);
+    operations.extend(qlty_config_operations(planner, &plugins)?);
+    operations.extend(fetch_operations(planner, &plugins));
```

Comment:
This approach of having 4 different functions which each produce operations seems OK for now, but I would watch it carefully to make sure it does not get out of hand.

I think the most important thing is to add code comments for these four functions which explain why the exist, to help keep it clear what each of the four functions is doing.